src/rgw/rgw_sal_motr.cc

```diff
@@ -2225,64 +2262,36 @@ int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir
         return rc;
       }
 
-      iter = bl.cbegin();
+      bufferlist& blr = bl;
+      iter = blr.cbegin();
       ent.decode(iter);
-      rc = 0;
       // Put into the cache
       this->store->get_obj_meta_cache()->put(dpp, this->get_name(), bl);
       goto out;
+  }
+  else
+  {
+    // Cache miss.
+    keys[0] = this->get_name() + "[";
 
-    }
-    else
-    {  // Version-id instance is empty
-       // Cache miss.
-        keys[0] = this->get_name();
-
-        // Retrieve all 'max' number of pairs.
-        rc = store->next_query_by_name(bucket_index_iname, keys, vals, this->get_name());
-        if (rc < 0) {
-          ldpp_dout(dpp, 0) << __func__ << "ERROR: NEXT query failed. " << rc << dendl;
-          return rc;
-        }
-
-        rc = -ENOENT;
-
-        // Iterating on object keys and return the latest object version
-        for (const auto& bl: vals) {
-          if (bl.length() == 0)
-            break;
-
-          iter = bl.cbegin();
-          ent.decode(iter);
-
-          if (ent.is_current()) {
-            ldpp_dout(dpp, 20) <<__func__<< ": found current version!" << dendl;
-            rc = 0;
-            // Put into the cache
-            this->store->get_obj_meta_cache()->put(dpp, this->get_name(), bl);
-            break;
-          }
-        }
-     }
-    } else {
-    if (this->store->get_obj_meta_cache()->get(dpp, this->get_name(), bl)) {
-      ldpp_dout(dpp, 20) <<__func__<< ": non-versioned bucket!" << dendl;
-
-      // TODO : Handle null version-id scenarios
+    // Retrieve all 'max' number of pairs.
+    rc = store->next_query_by_name(bucket_index_iname, keys, vals, this->get_name());
+    if (rc < 0) {
+      ldpp_dout(dpp, 0) << __func__ << "ERROR: NEXT query failed. " << rc << dendl;
+      return rc;
+      } 
 
-      rc = this->store->do_idx_op_by_name(bucket_index_iname,
-                                          M0_IC_GET, this->get_key().to_str(), bl);
-      if (rc < 0) {
-        ldpp_dout(dpp, 0) << __func__ << "ERROR: failed to get object's entry from bucket index: rc="
-                          << rc << dendl;
-        return rc;
+    if(rc == 0){
```

Comment:
rc == 0 is error here ??

Reply:
No, the motr call is successful. But, no entry found in the bucket.